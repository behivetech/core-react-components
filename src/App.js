import React, { Component } from 'react';
import FormField from './FormField';
import FormTextField from './FormTextField';
import Form from './Form';

export default class App extends Component {
    render() {
        return (
            <section className="app">
                <h1>Form Components</h1>
                <Form name="mockName">
                	<FormField name="mockFieldName1" label="Mock Field 1" helperText="test helper text 1" component="TextField" />
                	<FormField name="mockFieldName2" label="Mock Field 2" helperText="test helper text 2" component="TextField" />
            	</Form>
            </section>
        );
    }
}
