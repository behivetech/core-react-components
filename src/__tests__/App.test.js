import App from "../App";

// TODO: need to figure out how to test MDCSwitch and MDCFormField
describe('App', () => {
    const getEvent = (checked = false) => { 
        return {
            stopPropagation: jest.fn(),
            target: {
                checked,
                value: 'mock event value',
            },
        }
    };

    afterEach(() => {
        jest.clearAllMocks();
    });

    it('should render with the correct props', () => {
        const wrapper = shallow(<App name="mockName" className="mock-class" defaultValue="mock value" />);

        expect(wrapper.debug()).toMatchSnapshot();
    });
});