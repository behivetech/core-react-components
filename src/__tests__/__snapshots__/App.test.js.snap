// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render with the correct props 1`] = `
"<section className=\\"app\\">
  <h1>
    Form Components
  </h1>
  <Form name=\\"mockName\\" onSubmit={[Function: noop]}>
    <FormField name=\\"field1\\" label=\\"Field 1\\" componentName=\\"TextField\\" helperText=\\"test\\" validate={{...}} />
    <FormField defaultValue=\\"value\\" name=\\"field2\\" label=\\"Field 2\\" componentName=\\"Checkbox\\" helperText=\\"test\\" />
    <FormField defaultValue=\\"value\\" name=\\"field3\\" label=\\"Field 3\\" componentName=\\"Switch\\" helperText=\\"test\\" />
    <button type=\\"submit\\">
      submit
    </button>
  </Form>
</section>"
`;
